pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -637,6 +618,98 @@ func (w *wlReconciler) setupWithManager(mgr ctrl.Manager) error {
 		Complete(w)
 }
 
+func needsDelayedTopologyUpdate(local, remote *kueue.Workload) bool {
+	if remote == nil || remote.Status.Admission == nil || local == nil || local.Status.Admission == nil {
+		return false
+	}
+
+	for _, remotePSA := range remote.Status.Admission.PodSetAssignments {
+		for _, localPSA := range local.Status.Admission.PodSetAssignments {
+			if localPSA.Name != remotePSA.Name {
+				continue
+			}
+
+			if remotePSA.TopologyAssignment != nil && localPSA.TopologyAssignment == nil {
+				return true
+			}
+
+			if localPSA.DelayedTopologyRequest != nil &&
+				*localPSA.DelayedTopologyRequest == kueue.DelayedTopologyRequestStatePending &&
+				remotePSA.TopologyAssignment != nil {
+				return true
+			}
+			break
+		}
+	}
+	return false
+}
+
+func (w *wlReconciler) syncReservingRemoteState(ctx context.Context, group *wlGroup, reservingRemote string, acs *kueue.AdmissionCheckState) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	needsTopologyUpdate := needsDelayedTopologyUpdate(group.local, group.remotes[reservingRemote])
+	needsACUpdate := acs.State != kueue.CheckStateRetry && acs.State != kueue.CheckStateRejected
+
+	if !needsTopologyUpdate && !needsACUpdate {
+		return nil
+	}
+
+	if err := workload.PatchAdmissionStatus(ctx, w.client, group.local, w.clock, func() (*kueue.Workload, bool, error) {
+		if needsTopologyUpdate {
+			updateDelayedTopologyRequest(group.local, group.remotes[reservingRemote])
+		}
+
+		if needsACUpdate {
+			if group.jobAdapter.KeepAdmissionCheckPending() {
+				acs.State = kueue.CheckStatePending
+			} else {
+				acs.State = kueue.CheckStateReady
+			}
+			// update the message
+			acs.Message = fmt.Sprintf("The workload got reservation on %q", reservingRemote)
+			// update the transition time since is used to detect the lost worker state.
+			acs.LastTransitionTime = metav1.NewTime(w.clock.Now())
+
+			workload.SetAdmissionCheckState(&group.local.Status.AdmissionChecks, *acs, w.clock)
+			// Set the cluster name to the reserving remote and clear the nominated clusters.
+			group.local.Status.ClusterName = &reservingRemote
+			group.local.Status.NominatedClusterNames = nil
+		}
+
+		return group.local, true, nil
+	}); err != nil {
+		log.V(2).Error(err, "Failed to patch workload", "workload", klog.KObj(group.local))
+		return err
+	}
+
+	if needsACUpdate {
+		w.recorder.Eventf(group.local, corev1.EventTypeNormal, "MultiKueue", acs.Message)
+	}
+
+	return nil
+}
+
+func updateDelayedTopologyRequest(local, remote *kueue.Workload) {
+	if remote == nil || remote.Status.Admission == nil || local == nil || local.Status.Admission == nil {
+		return
+	}
+
+	for _, remotePSA := range remote.Status.Admission.PodSetAssignments {
+		for j := range local.Status.Admission.PodSetAssignments {
+			if local.Status.Admission.PodSetAssignments[j].Name != remotePSA.Name {
+				continue
+			}
```

Comment:
Please commonize the code for getting the PodSetAssignment by name and reuse the same helper in here: https://github.com/kubernetes-sigs/kueue/pull/5361/files#r2486006884